### PR TITLE
BLD: epicscorelibs pin matrix

### PR DESCRIPTION
**Pull request by ZLLentz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Previously, the most recent build of `pyca` silently required exactly `epicscorelibs=7.0.6.99.2.0` to import. Without this PR, each future build will silently require whichever most recent version of `epicscorelibs` that is available at the tagging time.

This PR accomplishes two things:

- Ensure that the recipe pins an exact version of `epicscorelibs` to use in both the build and run requirements. It needs to match exactly for the runtime import to succeed. This was always a silent requirement, so this makes it explicit and allows the solver to help you.
- Build with the current 3 most recent versions of `epicscorelibs` so that consumers have some flexibility.

This recipe will periodically need to be updated if `epicscorelibs` is updated. This can be accomplished by adding/removing versions from the `conda_build_config.yaml` file. This should be done on request and on `pyca` tag.

I tested this offline by building the package and verifying that the output has the correct dependencies and passes the `pyca` unit tests.

I've also removed Zrylettc from the maintainers list because he's moved on to other projects.

### CODE REVIEW

**Comment by ZLLentz** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -61,4 +61,3 @@ extra:
   recipe-maintainers:
     - klauer
     - zllentz
-    - zrylettc
```
Comment (klauer) [inline recipe/meta.yaml:64]:
```suggestion
    - tangkong
```

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+epicscorelibs:
```
Comment (klauer) [inline recipe/conda_build_config.yaml:1]: This is so unexpectedly easy. Thanks for figuring this out!

**Review by klauer — APPROVED** [review]
LGTM 👏

**Comment by ZLLentz** [comment]
This may need some further work, the first few CI jobs failed on some python incompatibility with epicscorelibs 7.0.4.99.1.2

**Comment by ZLLentz** [comment]
It looks like `epicscorelibs` v7.0.4.99.1.2 is not compatible with python versions above 3.9. The builds relating to these combinations failed. The obvious options are to either drop that build or just let those combinations fail.

One additional job failed with an HTTP error.

I'll take the suggested maintainers change, let the CI churn, and then merge as-is to see what happens unless there are other opinions.

File: recipe/meta.yaml (hunk 1 above)
Comment (ZLLentz) [reply recipe/meta.yaml:64]: Interestingly, you can't commit suggestions from the github ui on deleted lines. TIL.

**Comment by ZLLentz** [comment]
@conda-forge-admin, please rerender

**Comment by ZLLentz** [comment]
I tried to deselect the incompatible versions in the way that seemed most obvious

**Comment by ZLLentz** [comment]
It looks like the selector simply removed all of the 7.0.4 builds. Should I revert, leave it as is, or just explicitly drop 7.0.4? @klauer

**Comment by klauer** [comment]
I vote to remove the selector and see if we can get some builds

(... but am happy with whatever you push to this PR because chances are you'll see this comment after you've already decided)

**Comment by ZLLentz** [comment]
Ok, with that previous approval, let's merge here and see if we get some working pyca builds on conda-forge

**Pull request merged**